sync2/client.go

```diff
@@ -49,6 +52,34 @@ func NewHTTPClient(shortTimeout, longTimeout time.Duration, destHomeServer strin
 	}
 }
 
+func (v *HTTPClient) Versions(ctx context.Context) ([]string, error) {
+	req, err := http.NewRequestWithContext(ctx, "GET", v.DestinationServer+"/_matrix/client/versions", nil)
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Set("User-Agent", "sync-v3-proxy-"+ProxyVersion)
+	res, err := v.Client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	if res.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("/versions returned HTTP %d", res.StatusCode)
+	}
+	defer res.Body.Close()
+	body, err := ioutil.ReadAll(res.Body)
+	if err != nil {
+		return nil, err
+	}
+	var parsedRes struct {
+		Result []string `json:"versions"`
+	}
```

Comment:
Why bother parsing when we do nothing with it? Just waiting for a 200 OK is enough.

Reply:
I'm sanity checking that you're receiving a proper Matrix /versions response and not just e.g. some Apache "it works" webpage.